lib/ansible/modules/network/panos/panos_security_policy.py

```diff
@@ -0,0 +1,516 @@
+        required: false
+        default: None
+    devicegroup:
+        description: >
+            Device groups are used for the Panorama interaction with Firewall(s). The group must exists on Panorama.
+            If device group is not define we assume that we are contacting Firewall.
+        required: false
+        default: None
+    commit:
+        description:
+            - commit if changed
+        required: false
+        default: true
+'''
+
+EXAMPLES = '''
+# permit ssh to 1.1.1.1
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'SSH permit'
+    description: 'SSH rule test'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['any']
+    application: ['ssh']
+    service: ['application-default']
+    hip_profiles: ['any']
+    action: 'allow'
+    commit: false
+
+# Allow HTTP multimedia only from CDNs
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'HTTP Multimedia'
+    description: 'Allow HTTP multimedia only to host at 1.1.1.1'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['content-delivery-networks']
+    application: ['http-video', 'http-audio']
+    service: ['service-http', 'service-https']
+    hip_profiles: ['any']
+    action: 'allow'
+    commit: false
+
+# more complex fictitious rule that uses profiles
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'Allow HTTP w profile'
+    log_start: false
+    log_end: true
+    action: 'allow'
+    antivirus: 'default'
+    vulnerability: 'default'
+    spyware: 'default'
+    url_filtering: 'default'
+    wildfire_analysis: 'default'
+    commit: false
+
+# deny all
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'DenyAll'
+    log_start: true
+    log_end: true
+    action: 'deny'
+    rule_type: 'interzone'
+    commit: false
+
+# permit ssh to 1.1.1.1 using panorama and pushing the configuration to firewalls
+# that are defined in 'DeviceGroupA' device group
+- name: permit ssh to 1.1.1.1 through Panorama
+  panos_security_policy:
+    ip_address: '10.5.172.92'
+    password: 'paloalto'
+    rule_name: 'SSH permit'
+    description: 'SSH rule test'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['any']
+    application: ['ssh']
+    service: ['application-default']
+    hip_profiles: ['any']
+    action: 'allow'
+    devicegroup: 'DeviceGroupA'
+'''
+
+RETURN = '''
+# Default return values
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
```

Comment:
Minor issue, not currently enforced by CI though may be soon, could you please move `ANSIBLE_METADATA` to before the DOCUMENTATION block. Thanks

The order is:
#!/usr/bin/perl
GPL
ANSIBLE_METADATA
DOCUMENTATION
EXAMPLES
RETURNS